src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -331,13 +331,13 @@ Namespace Microsoft.CodeAnalysis.Semantics
             Return New LazyNameOfExpression(argument, _semanticModel, syntax, type, constantValue)
         End Function
 
-        Private Function CreateBoundLambdaOperation(boundLambda As BoundLambda) As ILambdaExpression
-            Dim signature As IMethodSymbol = boundLambda.LambdaSymbol
+        Private Function CreateBoundLambdaOperation(boundLambda As BoundLambda) As IAnonymousFunctionExpression
+            Dim symbol As IMethodSymbol = boundLambda.LambdaSymbol
             Dim body As Lazy(Of IBlockStatement) = New Lazy(Of IBlockStatement)(Function() DirectCast(Create(boundLambda.Body), IBlockStatement))
             Dim syntax As SyntaxNode = boundLambda.Syntax
             Dim type As ITypeSymbol = boundLambda.Type
```

Comment:
Unrelated to this change: can we put an assert here and simply set type to `Nothing`?